src/efi.rs

```diff
@@ -321,10 +321,16 @@ impl Component for Efi {
         log::debug!("Found metadata {}", meta.version);
         let srcdir_name = component_updatedirname(self);
         let ft = crate::filetree::FileTree::new_from_dir(&src_root.sub_dir(&srcdir_name)?)?;
-        // get esp device via /dev/disk/by-partlabel
-        let esp_device = self
-            .get_esp_device()
-            .ok_or_else(|| anyhow::anyhow!("Failed to find ESP device"))?;
+
+        // Get esp device via /dev/disk/by-partlabel
+        // If failed, change to sfdisk
+        let esp_device = if let Some(esp) = self.get_esp_device() {
+            esp
+        } else {
+            let esp = blockdev::get_esp_partition(device)?
```

Comment:
I wonder what the downside would be of just changing `self.get_esp_device()` to `self.get_esp_device(device)` and just use `blockdev::get_esp_partition(device)` inside of that function and not worry about `/dev/disk/by-label` at all.

It appears this is the only use of `self.get_esp_device()` that I see at this time.

Even better, get rid of `self.get_esp_device()` altogether and just use blockdev here:

```
let found_esp = blockdev::get_esp_partition(device)?                    
    .ok_or_else(|| anyhow::anyhow!("Failed to find ESP device"))?;
let esp = PathBuf::from(found_esp)                                                

```

of course, the variable names could be improved :)

Reply:
Doing this would also mean that we wouldn't need a huge comment explaining why we use `/dev/disk/by-label` in one case and inspect the partitions in another.